controllers/reconcile_persistence.go

```diff
@@ -117,6 +118,13 @@ func (r *TemporalClusterReconciler) reconcilePersistence(ctx context.Context, cl
 		return 0, fmt.Errorf("can't reconcile schema script configmap: %w", err)
 	}
 
+	// Ensure the serviceaccount used by jobs is up-to-date
+	serviceAccountBuilder := base.NewServiceAccountBuilder(persistence.ServiceNameSuffix, cluster, r.Scheme, &v1beta1.ServiceSpec{})
```

Comment:
This feels a little hacky, passing in the null ServiceSpec - I'll take a look at making the SA builder a little more generic

Reply:
The SA builder wasn't actually using the ServiceSpec at all so I've just removed it